classes/DataWarehouse/Visualization/AppKernelChart.php

```diff
@@ -0,0 +1,1075 @@
+                    $yMax=max($yMax, max($dataset->valueHighVector));
+                }
+                if(count($dataset->controlMaxVector)>0) {
+                    $yMax=max($yMax, max($dataset->controlMaxVector));
+                }
+                $dy=$yMax-$yMin;
+
+                $yMin-=0.05*$dy;
+                $yMax+=0.05*$dy;
+                if($yMin<0) {
+                    $yMin=0;
+                }
+
+                $yAxisColorValue = $colors[$this->_axisCount % 33];
+                $yAxisColor = '#'.str_pad(dechex($yAxisColorValue), 6, '0', STR_PAD_LEFT);
+                $yIndex = $this->_axisCount + 1;
+                $yAxisName = $this->_axisCount == 0 ? 'yaxis' : "yaxis{$yIndex}";
+                $yAxis = array(
+                    'automargin' => true,
+                    'layer' => 'below traces',
+                    'title' => array(
+                        'text' => '<b>' . $dataset->metricUnit . '</b>',
+                        'font' => array(
+                            'color'=> $yAxisColor,
+                            'size' => (12 + $font_size)
+                        )
+                    ),
+                    'tickfont' => array(
+                        'size' => (11 + $font_size),
+                        'color' => '#606060'
+                    ),
+                    'tickformat' => ',.2f',
+                    'tickmode' => 'auto',
+                    'ticksuffix' => ' ',
+                    'tickprefix' => $this->_axisCount > 0 ? ' ' : null,
+                    'exponentformat' => 'SI',
+                    'type' => 'linear',
+                    'rangemode' => 'tozero',
+                    'range' => [$yMin, $yMax],
+                    'index' => $yIndex,
+                    'axisname' => $yAxisName,
+                    'separatethousands' => true,
+                    'overlaying' => $this->_axisCount == 0 ? null : 'y',
+                    'linewidth' => 1 + $font_size/4,
+                    'linecolor' => '#c0d0e0',
+                    'side' => 'left',
+                    'anchor' => 'x',
+                    'autoshift' => true,
+                    'gridwidth' => $this->_axisCount > 1 ? 0 : 1 + ($font_size/8),
+                    'zeroline' => false,
+                );
+
+                if ($this->_axisCount > 0){
+                    if ($this->_axisCount % 2 == 0) {
+                        $yAxis = array_merge($yAxis, array(
+                            'side' => 'left',
+                            'anchor' => 'free',
+                            'autoshift' => true,
+                        ));
+                    }
+                    else {
+                        $yAxis = array_merge($yAxis, array(
+                            'side' => 'right',
+                            'anchor' => $this->_axisCount > 1 ? 'free' : 'x',
+                            'autoshift' => true,
+                        ));
+                    }
+                }
```

Comment:
Duplicate of:
https://github.com/ubccr/xdmod/pull/1822#discussion_r1586582420
Would be good to put in a function.